src/DSManager.cpp

```diff
@@ -794,7 +794,8 @@ void DSManager::loadClient(
     }
 
     // we must assert that DiscoveryProtocol is CLIENT
-    if (atts.rtps.builtin.discovery_config.discoveryProtocol != DiscoveryProtocol_t::CLIENT)
+    if (atts.rtps.builtin.discovery_config.discoveryProtocol != DiscoveryProtocol_t::CLIENT &&
+        atts.rtps.builtin.discovery_config.discoveryProtocol != DiscoveryProtocol_t::SUPER_CLIENT)
```

Comment:
We need some guards here for backwards compatibility. `SUPER_CLIENT` will only be from 2.3, but this should run for older branches too. Look at #27